autobotAI_integrations/integrations/gitlab/__init__.py

```
@@ -157,3 +159,20 @@ def generate_cli_creds(self) -> CLICreds:
         )
+
+    @staticmethod
+    def _is_public_gitlab_host(base_url: str) -> bool:
+        parsed = urlparse(str(base_url or ""))
+        return parsed.scheme.lower() == "https" and (parsed.hostname or "").lower() == "gitlab.com"
+
+    def generate_mcp_creds(self) -> MCPCreds:
+        # Autobot GitLab integrations authenticate with a user-provided PAT.
+        # MCP uses the same token; host is restricted to exact gitlab.com.
+        if not self._is_public_gitlab_host(self.integration.base_url):
+            raise Exception("Remote MCP is only supported for public gitlab.com")
```

Ruff (0.16.0)

[warning] Create your own exception

(TRY002)